subsys/net/ip/net_pkt.c

```diff
@@ -1474,8 +1474,14 @@ static inline bool insert_data(struct net_pkt *pkt, struct net_buf *frag,
 	do {
 		u16_t count = min(len, net_buf_tailroom(frag));
 
-		/* Copy insert data */
-		memcpy(frag->data + offset, data, count);
+		if (data) {
+			/* Copy insert data */
+			memcpy(frag->data + offset, data, count);
+		} else {
+			/* If there is no data, just clear the area */
+			memset(frag->data + offset, 0, count);
+		}
+
```

Comment:
I would have kept it as a one liner:
memcpy(..., data ? data : 0, count);

Reply:
2nd line is memset() tho ;-)

Reply:
Oh yes and memcpy of a plain byte value and not a pointer... time to get a break.